pi5/hw_pwm.go

```diff
@@ -24,17 +25,39 @@ import (
 // microsecond. We'll use 1 millisecond, for added confidence that all boards should support it.
 const safePeriodNs = 1e6
 
+/*
+These modes are used during pin control. Depending on which mode we'd like a pwm pin to
+be in (either GPIO or PWM), we overwrite a pin's mode data with one of these values.
+
+Note: Based on my (Maria's) experimentation, I'm pretty sure that ALT5 is the 'default' mode that tells the processor not
+to use alternative modes at all. In the pi5 library, they denote that SYS_RIO = ALT5, and my assumption here is
+that RIO = regular input output. regardless of whether I set a pin's mode to IP or OP, it the byte is still set to ALT5.
+GPIO usage uses different banks, addresses, registers, etc, so I'm pretty sure this is a way of telling the processor it
+shouldn't be using the alternative mode data at all.
+*/
+const (
+	ALT3 byte = 0x03 // PWM MODE
+	ALT5 byte = 0x05 // GPIO MODE
+
+	PWMMode  byte = ALT3
+	GPIOMode byte = ALT5
+)
+
 type pwmDevice struct {
 	chipPath string
 	line     int
 
 	// We have no mutable state, but the mutex is used to write to multiple pseudofiles atomically.
 	mu     sync.Mutex
 	logger logging.Logger
+
+	// virtual page that maps to memory associated with gpiochip0. We will overwrite bytes
+	// here to change a pin's mode from GPIO mode to HWPWM mode.
+	gpioChipVPage *mmap.MMap
```

Comment:
the virtual page stores the entirety of data related to the gpio chip, so it can be used for things other than switching pin modes. I feel like 'pin mode page' isn't quite right either. I'll change the name to gpioChipPage & update the comment.